salesforce-gui/src/main/java/org/fundacionjala/salesforce/utils/PageTransporter.java

```diff
@@ -1,14 +1,32 @@
 package org.fundacionjala.salesforce.utils;
 
 import org.fundacionjala.core.selenium.interaction.WebDriverManager;
-import org.fundacionjala.core.throwables.PropertiesReadingException;
 import org.fundacionjala.salesforce.config.SalesforceProperties;
+import org.openqa.selenium.support.ui.ExpectedConditions;
 
 import java.net.MalformedURLException;
 import java.net.URL;
 
+/**
+ * [MR] Class in charge of transport the WebDriver to a specific Url.
+ */
 public final class PageTransporter {
 
+    private static String instanceUrl;
+
+    static String getInstanceUrl() {
+        if (instanceUrl == null) {
+            instanceUrl = initInstanceUrl();
+        }
+        return instanceUrl;
+    }
+
+    private static String initInstanceUrl() {
+        WebDriverManager.getInstance().getWebDriverWait().until(
+                ExpectedConditions.not(ExpectedConditions.urlContains("login")));
+        return WebDriverManager.getInstance().getWebDriver().getCurrentUrl().split("\\.")[0].replace("--c", "");
```

Comment:
I think these lines can be reduced and add validations for a possible index exception.

instance = WebDriverManager.getInstance();
instance.getWebDriverWait().until( ExpectedConditions.not(ExpectedConditions.urlContains("login")));

String[] url =  instance.getWebDriver().getCurrentUrl().split("\\.");
if (url.length > 0 ){
   return url[0].replace("--c", "");
} else {
  return  throw new urlException();
}